src/sonic-config-engine/minigraph.py

```diff
@@ -98,7 +99,92 @@ def parse_device(device):
             deployment_id = node.text
     return (lo_prefix, lo_prefix_v6, mgmt_prefix, name, hwsku, d_type, deployment_id)
 
-def parse_png(png, hname):
+def calculate_lcm_for_ecmp (nhdevices_bank_map, nhip_bank_map):
+    banks_enumerated = {}
+    lcm_array = []
+    for value in nhdevices_bank_map.values():
+        for key in nhip_bank_map.keys():
+            if nhip_bank_map[key] == value:
+                if value not in banks_enumerated:
+                    banks_enumerated[value] = 1
+                else:
+                    banks_enumerated[value] = banks_enumerated[value] + 1
+    for bank_enumeration in banks_enumerated.values():
+        lcm_list = range(1, bank_enumeration+1)
+        lcm_comp = lcm_list[0]
+        for i in lcm_list[1:]:
+            lcm_comp = lcm_comp * i / calculate_gcd(lcm_comp, i)
```

Comment:
calculate_gcd can it return <=0 ? can we protect against that

Reply:
I believe gcd should be >0

L 497-530 should ensure that we only get to this portion of the code only if there are next hop ips for both ipv4 and ipv6 which would contribute to gcd value